### ISSUE TITLE
Question: Can I use LFS as an artifact repo?

### ISSUE THREAD

**Issue by Ricardo1980** [git-lfs/git-lfs#4441]
Hello,

I'm wondering if this is possible:

Can I use LFS as an artifact repo inside my project repo?

I mean, imagine every day I generate (using another repo) and upload a file (same filename, different content, let's say it is a framework) into my repo, a special branch.
Every time my CI uploads this file using LFS to github servers, we can print in the release notes the OID.
Because the file (different versions) is already uploaded by the CI, I don't want to upload them again once I create a feature branch and I want to use one of those files. So, can I do something, for example, editing a text file (the reference) and change the OID manually so a specific branch references a specific file version? (different feature branches use different versions of my framework file).

It seems to me that all those hooks and commands hide the OID complexity very well and although clearly it is supposed I shouldn't do what I say, I wonder if it is technically possible.

In the case that is possible, can I do that using 2 repos? one for the artifacts (framework) and another one for my app?
In other words, can the repo B reference a LFS file that is in repo A? (if this is not possible, that's not a problem, I can use only one repo and save the artifacts in a special branch)

Thanks a lot for your suggestions and ideas.

**Comment by bk2204** [comment]
Hey,

In general, it is possible to use Git LFS as an artifact store, although that's not a recommended use case for either Git or Git LFS.  Repositories, even repositories using Git LFS, aren't really meant for this purpose.  As to whether an object in one repository can refer to another, no, that's not possible.

Regardless of whether it's possible, it's not something that GitHub is going to want you to do.  GitHub doesn't want repositories that are data dumps or artifact stores because that results in repositories which are difficult to maintain.  I can't speak for other hosting providers, but I suspect they aren't going to like that, either.  So while you may physically be able to do so, you may be asked to move your repository elsewhere if you do.

You're better off storing these artifacts in either a release or a separate artifact store.
